R/images.R

```diff
@@ -0,0 +1,204 @@
+  ...) {
+  if (!is.function(device)) {
+    stop(call. = FALSE, "The `device` argument should be a function, e.g. `grDevices::png`")
+  }
+
+  expr <- rlang::enquo(expr)
+
+  tempFile <- missing(filename)
+
+  args <- rlang::list2(width = width, height = height, res = res)
+  argnms <- names(formals(device))
+  if (!"..." %in% argnms) {
+    # Only include `width`, `height`, and `res` if the corresponding formal
+    # parameters are present.
+    args <- args[names(args) %in% argnms | names(args) == "filename"]
+  }
+  args <- c(list(filename = filename), args, rlang::list2(...))
+
+  do.call(device, args)
+  dev <- grDevices::dev.cur()
+  on.exit(grDevices::dev.off(dev), add = TRUE, after = FALSE)
+
+  op <- graphics::par(mar = rep(0, 4))
+  # Prevent examples() from prompting
+  grDevices::devAskNewPage(FALSE)
+  tryCatch(graphics::plot.new(), finally = graphics::par(op))
+
+  tryCatch({
+    result <- withVisible(rlang::eval_tidy(expr))
+    if (result$visible) {
+      capture.output(print(result))
+    }
+    filename
+  }, error = function(e) {
+    try({
+      if (tempFile && file.exists(filename))
+        unlink(filename)
+    })
+    stop(e)
+  })
+}
+
+#' Capture a plot as a self-contained \code{<img>} tag
+#'
+#' @param expr A plotting expression that generates a plot (or yields an object
+#'   that generates a plot when printed, like a ggplot2).
+#' @param alt A single-element character vector that contains a text description
+#'   of the image. This is used by accessibility tools, such as screen readers
+#'   for vision impaired users.
+#' @param device A graphics device function; by default, this will be either
+#'   \code{\link[grDevices:png]{grDevices::png()}},
+#'   \code{\link[ragg:agg_png]{ragg::agg_png()}}, or
+#'   \code{\link[Cairo:CairoPNG]{Cairo::CairoPNG()}}, depending on your system
+#'   and configuration. See \code{\link{defaultPngDevice}}.
+#' @param width,height The width/height that the generated tag should be
+#'   displayed at, in logical (browser) pixels.
+#' @param pixelratio Indicates the ratio between physical and logical units of
+#'   length. For PNGs that may be displayed on high-DPI screens, use \code{2};
+#'   for graphics devices that express width/height in inches (like
+#'   \code{\link[grDevices:svg]{grDevices::svg()}}), try \code{1/72} or
+#'   \code{1/96}.
+#' @param mimeType The MIME type associated with the \code{device}. Examples are
+#'   \code{image/png}, \code{image/tiff}, \code{image/svg+xml}.
+#' @param deviceArgs A list of additional arguments that should be included when
+#'   the \code{device} function is invoked.
+#' @param attribs A list of additional attributes that should be included on the
+#'   generated \code{<img>} (e.g. \code{id}, \code{class}).
+#' @param suppressSize By default, \code{plotTag} will include a \code{style}
+#'   attribute with \code{width} and \code{height} properties specified in
+#'   pixels. If you'd rather specify the image size using other methods (like
```

Comment:
> By default, \code{plotTag} will include a \code{style} attribute with \code{width} and \code{height} properties specified in pixels.

Is this true? Looks to me as though there is no `style` attribute by default?

Reply:
```
> plotTag(plot(cars), alt = "foo")$attribs$style
[1] "width:400px;height:400px;"
```